test/NuGet.Tests.Apex/NuGet.Tests.Apex.Daily/NuGetUITestCase.cs

```diff
@@ -497,6 +497,106 @@ public async Task Uninstall_WithMultiplePackagesThatDependOnEachOther_PackageGoe
             uiwindow.AssertPackageNameAndType(transitivePackageName, NuGet.VisualStudio.PackageLevel.Transitive);
         }
 
+        [TestMethod]
+        [Timeout(DefaultTimeout)]
+        public async Task SearchTopLevelPackageInInstalledTabFromUI()
+        {
+            // Arrange
+            var transitivePackageName = "Contoso.B";
+            await CommonUtility.CreateDependenciesPackageInSourceAsync(_pathContext.PackageSource, TestPackageName, TestPackageVersionV1, transitivePackageName, TestPackageVersionV1);
+
+            NuGetApexTestService nugetTestService = GetNuGetTestService();
+
+            var solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            var project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.NetCoreClassLib, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
+
+            var uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
+            solutionService.Build();
+
+            CommonUtility.AssertPackageReferenceExists(VisualStudio, project, TestPackageName, TestPackageVersionV1, Logger);
+            uiwindow.AssertPackageNameAndType(TestPackageName, NuGet.VisualStudio.PackageLevel.TopLevel);
+            uiwindow.AssertPackageNameAndType(transitivePackageName, NuGet.VisualStudio.PackageLevel.Transitive);
```

Comment:
I'm not clear on why this assertion is made both before the `SearchPackageFromUI` call and after the call. Is there a common bug where installing shows the TopLevel package but a following Search does not show that TopLevel package?

Should `uiwindow.AssertPackageNameAndType(transitivePackageName, NuGet.VisualStudio.PackageLevel.Transitive);` also be asserted after the Act?

Reply:
Thanks for your review. 😊
First question: We call these assertions before `SearchPackageFromUI` to make sure that the package is installed successfully with the top-level package under the “top-level packages” node and the transitive packages under the “transitive packages” node.

The second question: the test case is search for packages under the “top-level packages” node, so after searching for top-level packages, only the top-level packages are displayed. So, we didn't add assert transitive.

Reply:
Makes sense. I believe also since the search is for an exact match package ID, there should be no transitive shown. 

It may be worth considering a test where you search "Contoso" and find both a top-level `Contoso.A` and a transitive `Contoso.C`, or similar.

Reply:
Thanks for your suggestion, we have added this test case `SearchTopLevelAndTransitivePackagePublicFieldInInstalledTabFromUI` to this PR, please help to review it, thanks again.